app/javascript/src/LoginPage/loginForms/Login3scaleForm.jsx

```diff
@@ -33,50 +39,40 @@ class Login3scaleForm extends React.Component<Props, State> {
     }
   }
 
-  setIsValidUsername = () => {
-    this.setState({isValidUsername: this.state.username !== ''})
-  }
-
-  handleTextInputUsername = (username: string) => {
-    this.setState({ username }, this.setIsValidUsername)
-  }
-
-  setIsValidPassword = () => {
-    this.setState({isValidPassword: this.state.password !== ''})
-  }
-
-  handleTextInputPassword = (password: string) => {
-    this.setState({ password }, this.setIsValidPassword)
+  handleInputChange = (value: string, event: SyntheticEvent<HTMLInputElement>) => {
+    const isValid = validateSingleField(event)
+    this.setState({
+      [namesToStateKeys[event.currentTarget.name].name]: value,
+      [namesToStateKeys[event.currentTarget.name].isValid]: isValid
+    })
   }
 
-  validateForm = (event: SyntheticEvent<HTMLInputElement>) => {
-    this.setIsValidUsername()
-    this.setIsValidPassword()
-    const isFormDisabled = this.state.username === '' ||
-      this.state.password === '' ||
-      this.state.isValidUsername === false ||
-      this.state.isValidPassword === false
-    if (isFormDisabled) {
+  validateForm = (event: SyntheticEvent<HTMLButtonElement>) => {
+    const errors = validateAllFields(event.currentTarget.form)
+    if (errors) {
       event.preventDefault()
+      errors.forEach(
+        (error) => this.setState({[namesToStateKeys[error].isValid]: false})
```

Comment:
I think errors is misnamed here. `error` is used like a name but it should be an error message. Maybe rename it to `invalidFields`? Or do `error => error.name ...`